### DISCUSSION TITLE
Pycharm plugin

### DISCUSSION THREAD

**Discussion by 9en9i** [astral-sh/ruff#9856] [Q&A] [answered]
Hello, about half a year ago pycharm introduced support for language server protocol. I saw that there is a plugin for vs code, as I think it works with ruff-lsp. I also know that there is a third-party plugin for pycharm, but it's a bit crude. And as you can assume official plugins are always better. Are there any plans to release an official plugin for pycharm?

**Comment by vivodi** [comment, marked as answer]
JetBrains officially provides Ruff support through PyCharm’s Python LSP plugin, and it is enabled by default. You don’t need to install any additional plugins. You can learn more at [JetBrains PyCharm LSP Tools documentation](https://www.jetbrains.com/help/pycharm/lsp-tools.html#ruff).
